cross-chain/arbitrum/deploy_l2/12_transfer_arbitrum_wormhole_gateway_ownership.ts

```diff
@@ -0,0 +1,18 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
```

Comment:
Before we transfer the ownership let's set [a self-reference to this gateway](https://github.com/keep-network/tbtc-v2/blob/e4a54b596629dfd57b0bb219867752f1b683f819/solidity/contracts/l2/L2WormholeGateway.sol#L354). I am not sure if Wormhole allows for hops from-to the same chain but I think it is less error-prone when we keep the mapping the same for all L2s.

Reply:
All our gateways [need to know the addresses](https://github.com/keep-network/tbtc-v2/blob/e4a54b596629dfd57b0bb219867752f1b683f819/solidity/contracts/l2/L2WormholeGateway.sol#L236) of other gateways. When we move tokens between chains, and there is no gateway, the token minted by Wormhole should be considered canonical. Addresses are stored [here](https://github.com/keep-network/tbtc-v2/blob/e4a54b596629dfd57b0bb219867752f1b683f819/solidity/contracts/l2/L2WormholeGateway.sol#L125). I think it makes sense if every gateway has the same mapping value, which includes adding a self-reference: current chain's ID as identified by Wormhole and the deployed gateway's address.

Reply:
Pls see [312e29b](https://github.com/keep-network/tbtc-v2/pull/554/commits/312e29b8c606a001a8ee4c93fb8fca68805d310d)